### DISCUSSION TITLE
duplicate recover; recover is turned on at line 26.

### DISCUSSION THREAD

**Discussion by lxzan** [scribble-rs/scribble.rs#324] [General]
https://github.com/scribble-rs/scribble.rs/blob/master/internal/api/ws.go#L26

https://github.com/scribble-rs/scribble.rs/blob/master/internal/api/ws.go#L148

**Comment by Bios-Marcel** [comment]
Yeah I haven't quite decided what to do when recovering. I was wondering whether its worth to force the player to reconnect, since the state could potentially be broken, as the player sent an event that might have been dropped without the player noticing.

I am not sure what the safest / cleanest solution would be.

**Comment by lxzan** [comment]
The panic is triggered by your business logic, it does not cause the connection to close.

I'd like to just print the error stack, and fix the problem.

**Reply by Bios-Marcel** [reply]
Yeah, I know. But if the business logic doesn't run, the overall application state (connection state) might be faulty. As the client thinks the server has received certain events which it hasn't. Currently there's nothing like an "ack" system, where the client first sends an event and only then modifies its local state.

Maybe I gotta rethink some stuff.
